controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -88,17 +88,72 @@ func (r *StorageAutoscalerReconciler) Reconcile(ctx context.Context, request rec
 
 	// if no expansion is in-progress update the phase to "not-started"
 	if storageAutoScaler.Status.Phase == "" {
-		originalStorageAutoScaler := storageAutoScaler.DeepCopy()
 		storageAutoScaler.Status.Phase = ocsv1.StorageAutoScalerPhaseNotStarted
-		err := r.updateStatus(ctx, storageAutoScaler, originalStorageAutoScaler)
+		err := r.updateStatus(ctx, storageAutoScaler)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	// if the expansion is InProgress or Failed, verify the scaling
-	if storageAutoScaler.Status.Phase == ocsv1.StorageAutoScalerPhaseInProgress || storageAutoScaler.Status.Phase == ocsv1.StorageAutoScalerPhaseFailed {
-		return r.verifyScaling(ctx, storageAutoScaler)
+	// get the current osd size and count from prometheus
+	currentOsdSize, currentOsdCount, err := r.getCurrentOsdSizeAndCount(ctx, storageAutoScaler)
```

Comment:
Long methods are difficult to maintain. Can you consider breaking out some of these new checks into helper methods?

Reply:
Please add unit tests, which will also be easier to add when there are smaller methods.

Reply:
Should I bring back the verify.go file??

Reply:
We don't need a separate file, just smaller helper methods. It's perfectly fine to have many methods in the same file if they are all related.